cmake/extra_fast.cmake

```diff
@@ -0,0 +1,28 @@
+include(CheckCXXCompilerFlag)
+
+# Enable agressive optimisation flags.
+
+# Try to use -march=x86-64-v3, but fall back to -march=native if we are
+# using an old compiler that doesn't support that flag.
+check_cxx_compiler_flag("-march=x86-64-v3" COMPILER_SUPPORTS_MARCH_X86_V3)
```

Comment:
Not sure this is the best default to use, but it's probably fine since it will fail on e.g. AArch64 hosts and the fall back to -march=native.

Reply:
This whole feature doesn’t belong here. It’s a separate change (and could equally have been done with make)

Reply:
Fair point, I'll defer it.